src/OrleansRuntime/Core/Dispatcher.cs

```diff
@@ -370,7 +375,20 @@ public void HandleIncomingRequest(Message message, ActivationData targetActivati
         {
             lock (targetActivation)
             {
-                if (targetActivation.State == ActivationState.Invalid)
+                if (targetActivation.Grain.IsGrain)
+                {
+                    var request = ((InvokeMethodRequest)message.GetDeserializedBody(this.serializationManager));
```

Comment:
Interface version alone will not be sufficient, we will need to move also `InterfaceId`. We don't use the `MethodId` for placement, so it could make sense to leave it here.

If we move `InterfaceId` and `InterfaceVersion` in headers, we will deserialize two additional intergers for each message received. Here we only deserialize the whole body for new placement and when the target activation is about to process the request (needed just after anyway, unless the current activation is incompatible).

So it is hard to know which implementation will be more efficient, as it will largely depends on the use case (lot of new activations with big params vs not so much new activations with small params)

THe best solution seems to move `InterfaceId` and `InterfaceVersion` into a new structure, included in the message payload, that would be deserialized only if needed.